types/collections.go

```diff
@@ -85,6 +85,37 @@ func (a genericAddressKey[T]) SizeNonTerminal(key T) int {
 	return collections.BytesKey.SizeNonTerminal(key)
 }
 
+type genericAddressIndexKey[T addressUnion] struct {
```

Comment:
can we get a godoc here

Reply:
the real doc is here (since it's exported): https://github.com/cosmos/cosmos-sdk/pull/15327/files#diff-88ec3984207abcf0ab89f34ad025d88346da2d91d516719e5cb342e0e0798059R104

I can add a pointer to it here.